modules/py_csi_ng.c

```diff
@@ -223,7 +223,11 @@ static mp_obj_t py_csi_snapshot(size_t n_args, const mp_obj_t *pos_args, mp_map_
 
         // If an image is provided update it and return.
         if (args[ARG_image].u_obj != mp_const_none) {
-            image_copy(py_helper_arg_to_image(args[ARG_image].u_obj, ARG_IMAGE_ANY), &image, true);
+            image_t *other = py_helper_arg_to_image(args[ARG_image].u_obj, ARG_IMAGE_MUTABLE);
+            fb_alloc_mark();
+            imlib_draw_image(other, &image, 0, 0, 1.f, 1.f, NULL, -1, 255, NULL, NULL,
+                             IMAGE_HINT_SCALE_ASPECT_IGNORE, NULL, NULL, NULL);
+            fb_alloc_free_till_mark();
```

Comment:
I assume this checks the bounds right? What's the fb_alloc for?

Reply:
imlib_draw_image does fb_alloc internally, so, it needs mark around it.

Yes, it checks bounds.